File: Ent_UsAdmin/Caja/RecibirCierre.cs
using System;
using System.ComponentModel;
using System.Data;
using System.Data.SqlClient;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace ConfiaAdmin
{

    public partial class RecibirCierre
    {

        public bool autorizado;
        public int idRetiro;
        public double Cantidad;
        private DataTable dataTickets;
        private SqlDataAdapter adapterTickets;
        private Cargando nCargando;

        public RecibirCierre()
        {
            InitializeComponent();
        }
        private void RecibirRetiro_Load(object sender, EventArgs e)
        {
            lblMonto.Text = Strings.FormatCurrency(Cantidad, 2);
            nCargando = new Cargando();
            nCargando.MonoFlat_Label1.Text = "Cargando";
            nCargando.Show();
            BackgroundDetalleCierre.RunWorkerAsync();
        }

        private void MonoFlat_Button1_Click(object sender, EventArgs e)
        {

            BackgroundWorker1.RunWorkerAsync();


        }

        private void BackgroundWorker1_DoWork(object sender, DoWorkEventArgs e)
        {
            Module1.iniciarconexionempresa();
            string fechainsercionhasta;
            double monto;
            string fechaRetiro;
            string concepto;
            monto = Conversions.ToDouble(txtmonto.Text);
            string tiempo = DateAndTime.TimeOfDay.ToString("HH:mm:ss");
            fechainsercionhasta = Conversions.ToString(DateTime.Now.Date);

            DateTime fechasqlhasta;
            fechasqlhasta = Conversions.ToDate(fechainsercionhasta);
            fechainsercionhasta = Strings.Format(fechasqlhasta, "yyyy-MM-dd");
            fechaRetiro = Strings.Format(fechasqlhasta, "dd/MM/yyyy");
            SqlCommand comandoRecibirRetiro;
            string consultaRetiro;
            consultaRetiro = "update cierrecajagestores set MontoRecibido = '" + txtmonto.Text + "',fechaRecibido = '" + fechainsercionhasta + "',usuariorecibe = '" + Module1.nmusr + "',horarecibido = '" + tiempo + "' where id = '" + idRetiro + "'";
            comandoRecibirRetiro = new SqlCommand();
            comandoRecibirRetiro.Connection = Module1.conexionempresa;
            comandoRecibirRetiro.CommandText = consultaRetiro;
            comandoRecibirRetiro.ExecuteNonQuery();

        }

        private void BackgroundWorker1_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {

            My.MyProject.Forms.Cargando.Close();
            Invoke(new Action(() => My.MyProject.Forms.CierresSinRecibir.CargarRetiros()));

            Close();
        }

        private void Panel1_Paint(object sender, PaintEventArgs e)
        {

        }

        private void Panel1_MouseDown(object sender, MouseEventArgs e)
        {
            if (e.Button == MouseButtons.Left)
            {
                Module_resize.MoveForm(this);
            }
        }

        private void BackgroundDetalleCierre_DoWork(object sender, DoWorkEventArgs e)
        {
            string consultaDetalleCierre;
            Module1.iniciarconexionempresa();
            consultaDetalleCierre = "select dc.id,dc.idTicket,dc.MontoTicket,Ticket.Fecha from detallecierrecajagestores dc inner join ticket on dc.idticket = ticket.id where dc.idcierre = '" + idRetiro + "'";
            adapterTickets = new SqlDataAdapter(consultaDetalleCierre, Module1.conexionempresa);
            dataTickets = new DataTable();
            adapterTickets.Fill(dataTickets);

        }

        private void BackgroundDetalleCierre_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {
            dtimpuestos.DataSource = dataTickets;
            nCargando.Close();
        }
    }
}

File: Ent_UsAdmin/RecibirCierre.vb
Imports System.ComponentModel
Imports System.Data.SqlClient

Public Class RecibirCierre

    Public autorizado As Boolean
    Public idRetiro As Integer
    Public Cantidad As Double
    Dim dataTickets As DataTable
    Dim adapterTickets As SqlDataAdapter
    Dim nCargando As Cargando
    Private Sub RecibirRetiro_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        lblMonto.Text = FormatCurrency(Cantidad, 2)
        nCargando = New Cargando
        nCargando.MonoFlat_Label1.Text = "Cargando"
        nCargando.Show()
        BackgroundDetalleCierre.RunWorkerAsync()
    End Sub

    Private Sub MonoFlat_Button1_Click(sender As Object, e As EventArgs) Handles MonoFlat_Button1.Click

        BackgroundWorker1.RunWorkerAsync()


    End Sub

    Private Sub BackgroundWorker1_DoWork(sender As Object, e As System.ComponentModel.DoWorkEventArgs) Handles BackgroundWorker1.DoWork
        iniciarconexionempresa()
        Dim fechainsercionhasta As String
        Dim monto As Double
        Dim fechaRetiro As String
        Dim concepto As String
        monto = txtmonto.Text
        Dim tiempo As String = TimeOfDay.ToString("HH:mm:ss")
        fechainsercionhasta = Now.Date

        Dim fechasqlhasta As Date
        fechasqlhasta = fechainsercionhasta
        fechainsercionhasta = Format(fechasqlhasta, "yyyy-MM-dd")
        fechaRetiro = Format(fechasqlhasta, "dd/MM/yyyy")
        Dim comandoRecibirRetiro As SqlCommand
        Dim consultaRetiro As String
        consultaRetiro = "update cierrecajagestores set MontoRecibido = '" & txtmonto.Text & "',fechaRecibido = '" & fechainsercionhasta & "',usuariorecibe = '" & nmusr & "',horarecibido = '" & tiempo & "' where id = '" & idRetiro & "'"
        comandoRecibirRetiro = New SqlCommand
        comandoRecibirRetiro.Connection = conexionempresa
        comandoRecibirRetiro.CommandText = consultaRetiro
        comandoRecibirRetiro.ExecuteNonQuery()

    End Sub

    Private Sub BackgroundWorker1_RunWorkerCompleted(sender As Object, e As RunWorkerCompletedEventArgs) Handles BackgroundWorker1.RunWorkerCompleted

        Cargando.Close()
        Me.Invoke(Sub()
                      CierresSinRecibir.CargarRetiros()
                  End Sub)

        Me.Close()
    End Sub

    Private Sub Panel1_Paint(sender As Object, e As PaintEventArgs) Handles Panel1.Paint

    End Sub

    Private Sub Panel1_MouseDown(sender As Object, e As MouseEventArgs) Handles Panel1.MouseDown
        If e.Button = MouseButtons.Left Then
            MoveForm(Me)
        End If
    End Sub

    Private Sub BackgroundDetalleCierre_DoWork(sender As Object, e As DoWorkEventArgs) Handles BackgroundDetalleCierre.DoWork
        Dim consultaDetalleCierre As String
        iniciarconexionempresa()
        consultaDetalleCierre = "select dc.id,dc.idTicket,dc.MontoTicket,Ticket.Fecha from detallecierrecajagestores dc inner join ticket on dc.idticket = ticket.id where dc.idcierre = '" & idRetiro & "'"
        adapterTickets = New SqlDataAdapter(consultaDetalleCierre, conexionempresa)
        dataTickets = New DataTable
        adapterTickets.Fill(dataTickets)

    End Sub

    Private Sub BackgroundDetalleCierre_RunWorkerCompleted(sender As Object, e As RunWorkerCompletedEventArgs) Handles BackgroundDetalleCierre.RunWorkerCompleted
        dtimpuestos.DataSource = dataTickets
        nCargando.Close()
    End Sub
End Class